### PR TITLE
1326-filtering-of-git-branches-in-the-selector

### PR DESCRIPTION
**Pull request by swordEdge**

### CODE REVIEW

**Comment by six7** [comment]
Not sure if this is ready for review yet, but:

- Can we change the logic so it only displays the search bar if there's more than 5 branches? Right now it displays even when there's just 1

There seems to be some padding issue on the right of the second dropdown that appears. Let's make this look like the first

image

There seems to be another padding issue with the `branch` item compared to the `Create new branch from..`
image
(notice the blue background doesnt span to the right)
image

File: src/app/components/BranchSelector.tsx [hunk 1]
```diff
@@ -184,12 +198,30 @@ export default function BranchSelector() {
...
   }, []);
 
+  const handleSearchTextForBaseBranchChange = React.useCallback<React.ChangeEventHandler<HTMLInputElement>>((e) => {
+    setSearchTextForBaseBranch(e.target.value);
+  }, []);
+
+  const handleSearchTextForCreateBranchChange = React.useCallback<React.ChangeEventHandler<HTMLInputElement>>((e) => {
+    setSearchTextForCreateBranch(e.target.value);
+  }, []);
+  // TODO: I'm not sure there is another way instead of stopImmediatePropagation to prevent option selection when typing the first letter of the option
```
Comment (six7) [inline src/app/components/BranchSelector.tsx:220]: this seems to select the first item for me as soon as i start typing 

image

**Review by six7 — CHANGES_REQUESTED** [review]

File: src/app/components/BranchSelector.tsx (hunk 1 above)
Comment (six7) [reply src/app/components/BranchSelector.tsx:220]: also got an error in console:

image

Comment (swordEdge) [reply src/app/components/BranchSelector.tsx:220]: Did you upgrade the node module? This error is happening in old version so upgrated it.

Comment (swordEdge) [reply src/app/components/BranchSelector.tsx:220]: What I'm not sure is that all test cases are passed with old version but failed with upgraded version.

Comment (six7) [reply src/app/components/BranchSelector.tsx:220]: You're right, this is gone now 👍

Comment (six7) [reply src/app/components/BranchSelector.tsx:220]: i'm still getting that error in the console, i did install, yes

image

**Review by six7 — CHANGES_REQUESTED** [review]
when you set figma to light mode it currently displays the search field with wrong colors. you can just use raw hex values for these instead, as the context menu has the same colors both in light and dark mode. but create a token for that, so that it's stored in stitches.config

image

**Comment by six7** [comment]
just want to check in on this one: is there anything i can help with? as far as i can see its just the error in the console that's still there, the rest is done 🙏

**Pull request closed without merging**